tests/s3/multipart_upload/test_multipart_upload.py

```diff
@@ -517,20 +524,31 @@ def test_multiprt_upload_having_more_than_10000_parts_2294(self):
         mpu_id = res[1]["UploadId"]
         self.log.info("Multipart Upload initiated with mpu_id %s", mpu_id)
         self.log.info("Uploading parts into bucket")
-        try:
+        if S3_ENGINE_RGW == CMN_CFG["s3_engine"]:
             resp = self.s3_mp_test_obj.upload_parts(
                 mpu_id,
                 self.bucket_name,
                 self.object_name,
                 mp_config["file_size"],
                 total_parts=mp_config["total_parts"],
                 multipart_obj_path=self.mp_obj_path)
-            assert_utils.assert_false(resp[0], resp[1])
-        except CTException as error:
-            self.log.error(error.message)
-            assert_utils.assert_in(errmsg.S3_MULTIPART_INVALID_PART_ERR, error.message,
-                                   error.message)
-        self.log.info("Cannot upload more than 10000 parts upload")
+            assert_utils.assert_true(resp[0], resp[1])
+            self.log.info("Uploaded more than 10000 parts")
+        else:
+            try:
+                resp = self.s3_mp_test_obj.upload_parts(
+                    mpu_id,
+                    self.bucket_name,
+                    self.object_name,
+                    mp_config["file_size"],
+                    total_parts=mp_config["total_parts"],
+                    multipart_obj_path=self.mp_obj_path)
+                assert_utils.assert_false(resp[0], resp[1])
+            except CTException as error:
```

Comment:
raising exception in tests?

Reply:
We are not raising here, we are just handling it. Whatever exception raised by lib, is handled here and we need it to catch the exact error